### DISCUSSION TITLE
Use content type hub

### DISCUSSION THREAD

**Discussion by Remi749** [Puzzlepart/prosjektportalen365#846] [General]
**Describe the solution you'd like**

Use content type hub instead of having a content type at Portfolio level that is disconnected at Project level. 

This will make introducing new fields, renaming, etc... easier for customers as everything will be done through the content type hub.

There are already plenty of customers that has made changes toe content types, and have had to create scripts to introduce them to existing projects.

Alternatively create a greneral script that works like UpgradeAllSitesToLatest that does this job for customers that has made changes to content types or fields.

**Comment by olemp** [comment]
This is a huge architectural change, and can't just be included in a patch version. And for customers with several project portals, how would it work? I don't think we should do this to be honest.

**Reply by Remi749** [reply]
I see and understand that. The customer sees this as high priority, but if this shouldn't be a change done to Prosjektportalen directly, it will be a custom change for that customer instead.

If more agrees then we'll close this discussion.

EDIT: Added an alternative to OP.

@okms , @pzljanb , @tarjeieo , thoughts?

**Reply by tarjeieo** [reply]
Agreed, a lot of impact and not that big of an upside tbh. CTH was considered back in the day, but it was quite buggy and unstable. Might be in a better state now?
